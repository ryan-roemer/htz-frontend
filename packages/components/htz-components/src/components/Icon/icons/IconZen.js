/* *************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change this Icon component , it is generated
 * from the `iconTamplate.js` file the parent directory.
 * *************************************************************** */
import React from 'react';
import { createComponent, } from 'react-fela';
import iconStyle from '../iconStyle';
import { iconPropTypes, iconDefaultProps, } from '../iconPropTypes';
import { attrsPropType, } from '../../../propTypes/attrsPropType';

const IconZen = createComponent(
  iconStyle,
  UnstyledIconZen,
  [ 'attrs', 'onClick', ]
);

IconZen.propTypes = iconPropTypes;
IconZen.defaultProps = iconDefaultProps;


// Underlying component
UnstyledIconZen.propTypes = {
  attrs: attrsPropType,
};

UnstyledIconZen.defaultProps = {
  attrs: null,
};

function UnstyledIconZen({ attrs, ...props }) {
  return (
    <svg width="1em" height="1em" viewBox="0 0 256 256" {...props} {...attrs} >  <path fill="currentColor" d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z" /></svg>
  );
}


export default IconZen;
