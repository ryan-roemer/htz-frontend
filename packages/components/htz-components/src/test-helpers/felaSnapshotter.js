import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
