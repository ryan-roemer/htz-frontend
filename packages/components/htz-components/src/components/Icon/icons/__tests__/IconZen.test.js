
/* global window */
/* *************************************************************** *
 * THIS IS AN AUTO GENERATED FILE. PLEASE DO NOT EDIT IT DIRECTLY.
 *
 * If you want to change the styleguide example, it is generated
 * from the `embedsFileTemplate.js` file is this directory.
 * *************************************************************** */
import React from 'react';
import felaSnapshotter from '../../../../test-helpers/felaSnapshotter';
import IconZen from '../IconZen';

describe('<IconZen />', () => {
  /* The error tests are left as documentation, but are commented out for the redability of the test results */
  // describe('Errors', () => {
  //   it('throw an error when the value passed to the "size" prop isn't a number', () => {
  //     expect(() => felaSnapshotter(<IconZen size="2rem" />)).toThrow(
  //       /^An Icon's "size" prop may only be passed a "number"/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconZen color="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "color" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconZen color={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (string)', () => {
  //     expect(() => felaSnapshotter(<IconZen fill="red" />)).toThrow(
  //       /is not a named-color/
  //     );
  //   });
  //   it('throw an error when the value passed to the "fill" prop isn't a named color (array of strings)', () => {
  //     expect(() =>
  //       felaSnapshotter(<IconZen fill={[ 'primary', 'nope', ]} />)
  //     ).toThrow(/is not a named-color/);
  //   });
  // });
  describe('DOM element', () => {
    testCase('render correctly when no props are passed', <IconZen />);
    testCase(
      'pass attributes defined in the "attrs" prop to the DOM element',
      <IconZen attrs={{ 'aria-hidden': true, }} />
    );
    testCase('add "id" attribute to DOM element', <IconZen id="hasId" />);
    testCase(
      'add "onClick" attribute to DOM element',
      <IconZen id={e => console.log('a')} />
    );
  });

  describe('styles', () => {
    testCase("set icon's color", <IconZen color={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's color responsively",
      <IconZen
        color={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's fill", <IconZen fill={[ 'primary', 'base', ]} />);
    testCase(
      "set icon's fill responsively",
      <IconZen
        fill={[
          { until: 'xl', value: [ 'primary', 'base', ], },
          { from: 'xl', value: [ 'bodyText', 'base', ], },
        ]}
      />
    );
    testCase("set icon's size", <IconZen size={6} />);
    testCase(
      "set icon's size responsively",
      <IconZen
        size={[ { until: 'xl', value: 6, }, { from: 'xl', value: 7, }, ]}
      />
    );
    testCase(
      'set misc styles on an icon',
      <IconZen miscStyles={{ position: 'relative', }} />
    );
    testCase(
      'responsively set misc styles on an icon',
      <IconZen
        miscStyles={{
          position: [
            { until: 'xl', value: 'relative', },
            { from: 'xl', value: 'static', },
          ],
        }}
      />
    );
  });
});

function testCase(name, Component, { only = false, } = {}) {
  const testFunc = only ? it.only : it;
  testFunc(name, () => {
    const { component, styles, } = felaSnapshotter(Component);
    expect(component).toMatchSnapshot();
    expect(styles).toMatchSnapshot();
  });
}
