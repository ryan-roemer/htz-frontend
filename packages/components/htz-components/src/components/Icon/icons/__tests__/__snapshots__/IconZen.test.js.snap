// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconZen /> DOM element add "id" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id="hasId"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> DOM element add "id" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconZen /> DOM element add "onClick" attribute to DOM element 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    id={[Function]}
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> DOM element add "onClick" attribute to DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconZen /> DOM element pass attributes defined in the "attrs" prop to the DOM element 1`] = `
<div>
  <svg
    aria-hidden={true}
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> DOM element pass attributes defined in the "attrs" prop to the DOM element 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconZen /> DOM element render correctly when no props are passed 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> DOM element render correctly when no props are passed 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}"`;

exports[`<IconZen /> styles responsively set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles responsively set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 80em){.htz-d{position:relative}}@media (min-width: 80em){.htz-e{position:static}}"`;

exports[`<IconZen /> styles set icon's color 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set icon's color 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:#0B7EB5}.htz-c{fill:#FFF}"`;

exports[`<IconZen /> styles set icon's color responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set icon's color responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-d{fill:#FFF}@media (max-width: 80em){.htz-b{color:#0B7EB5}}@media (min-width: 80em){.htz-c{color:#2D2D2D}}"`;

exports[`<IconZen /> styles set icon's fill 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set icon's fill 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#0B7EB5}"`;

exports[`<IconZen /> styles set icon's fill responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set icon's fill responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}@media (max-width: 80em){.htz-c{fill:#0B7EB5}}@media (min-width: 80em){.htz-d{fill:#2D2D2D}}"`;

exports[`<IconZen /> styles set icon's size 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set icon's size 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{font-size:6rem}"`;

exports[`<IconZen /> styles set icon's size responsively 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d htz-e"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set icon's size responsively 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}@media (max-width: 80em){.htz-d{font-size:6rem}}@media (min-width: 80em){.htz-e{font-size:7rem}}"`;

exports[`<IconZen /> styles set misc styles on an icon 1`] = `
<div>
  <svg
    className="htz-a htz-b htz-c htz-d"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
  >
      
    <path
      d="M128 8C61.7 8 8 61.7 8 128s53.7 120 120 120 120-53.7 120-120S194.3 8 128 8zm17 188h-34V60h34v136z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`<IconZen /> styles set misc styles on an icon 2`] = `".htz-a{vertical-align:-0.15em}.htz-b{color:inherit}.htz-c{fill:#FFF}.htz-d{position:relative}"`;
